.reek.yml

```diff
@@ -125,7 +125,7 @@ detectors:
   UtilityFunction:
     enabled: true
     exclude: []
-    public_methods_only: false
+    public_methods_only: true
```

Comment:
habilita para privado e protegido ? se for:

https://github.com/troessner/reek/blob/master/docs/Utility-Function.md

Não entendo porque excluir dos metodos privados e protegidos.

Reply:
Habilita para privado e protegido porque você poderia fazer funções 'utilitárias' em serviços internos. Vamos dizer que você vai fazer um método que vai encapsular o que você está recebendo de uma outra forma, como:

```
def preciso_tratar(hash_interna)
  id: hash_interna.id, nome: hash_interna.name }
end
```

Este método é considerado utility porque não depende de nada do estado interno do serviço, por exemplo. Entendo que é possível escrever serializers ou afins para evitar o cop, mas vamos fazer isto para todos os métodos que porventura tivermos de passar o parâmetro? (para evitar artificialmente o cop, você pode atribuir um `attr_accessor` no seu serviço e usar ele na ordem, mas te obriga a fazer desta forma que, dependendo da complexidade, pode levar a erros)

Reply:
Daí, estando em método privado, sabemos que este utility estará apenas para instâncias daquela classe (ou, em protegidos, em herdeiros, enfim)